tests/network/macvtap.go

```diff
@@ -87,184 +78,21 @@ var _ = SIGDescribe("Macvtap", decorators.Macvtap, Serial, func() {
 			chosenMACHW, err := GenerateRandomMac()
 			Expect(err).ToNot(HaveOccurred())
 			chosenMAC = chosenMACHW.String()
-			serverCIDR := "192.0.2.102/24"
 
-			serverIP, err = libnet.CidrToIP(serverCIDR)
+			const macvtapNetName = "test-macvtap"
+			serverVMI := libvmi.NewAlpineWithTestTooling(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(v1.DefaultMacvtapNetworkInterface(macvtapNetName), chosenMAC)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(macvtapNetName, macvtapNetworkName)),
+				libvmi.WithNodeAffinityFor(nodeName),
```

Comment:
Why should the VM be pinned to a specific node?

Reply:
To be honst I dont know its seem redundant, the reason I did not removed it, and some other chanags around the VM creation helpers, is this PR got bigger then I anticipated and I prefer to not add more to it.

Do you mind if I do these changes in a follow up PR?

Reply:
I removed the node affinity.

As I said I think its redundant because it would set affinity for scheduable nodes.
It is expected that the scheduler figure out the best nodes for the VMs.

Reply:
I reverted, it seems we need the node affinity to schedule the VM on the same node otherwise the test fails.
Maybe its related to how the tests cluster is create on CI, anyhow this is not in this PR scope to investigate it.

Reply:
> I reverted, it seems we need the node affinity to create the VM on the same node otherwise the test fails, probably due to missing wiring between the nodes macvtap interfaces.

The same node as what? The test is not testing connectivity to anything.
Since we are deprecation it anyway, if you don't know the answer, don't spend too much time on it. Not a blocking comment...

Reply:
So both VMs will create on the same node.
The test checks connectivity between two VMs, on the same node, over the macvtap network.

Reply:
Maybe I miss something, but the test here is not testing connectivity and doesn't have two VMs. It runs a singe VMI and check it mac.
But I see the final version of this metod is fine and the node affinity was removed.

Reply:
Sorry for the confusion, this is another test as you realized, yes its final form is having no node affinity.